### DISCUSSION TITLE
Enable airflow metrics with statsd

### DISCUSSION THREAD

**Discussion by mingmwang** [apache/airflow#34266] [Q&A]
When I change the airflow config to enable the metrics with statsd, with both the airflow `scheduler` and `executors` push metrics to the statsd-exporter or just the `scheduler` ?

```
statsd_on = True
statsd_host = localhost
statsd_port = 8125
statsd_prefix = airflow
```

**Comment by potiuk** [comment]
Executor is generally running as part of scheduler and statsd configuration enables "general" enabling. All the configuration parameters  can be found in https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#metrics

BTW. We recommend to switch (in the newer Airflow 2.6+ releases ) to use Open Telemetry metrics integration instead of statsd. It is something that Airflow will improve in the future (there is a talk planned on this next week at the Airflow Summit in Toronto https://airflowsummit.org/sessions/2023/apache-airflow-and-opentelemetry/)

**Comment by mingmwang** [comment]
@potiuk 

The executors I mean is actually the workers, might be `CeleryExecutor` or `DaskExecutor` in real deployment.
Based the below architecture graph, it seems only the scheduler will push metrics to statsd, please help to confirm.

https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-architecture.html

**Reply by potiuk** [reply]
Just looka at available metrics. Different components of airflow have different metrics - I do not remember which one is produced by which component - but you should look at the available metrics and find out. All airflow components (See the docs) should have the same configuration and the  single configuration controls it for all of them.

**Reply by potiuk** [reply]
List of metrics here https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/metrics.html#rename-metrics

**Reply by cris4414** [reply]
Where can I find the implementation of all the metrics in the above documentation link on Airflow Github?
Thanks

**Reply by potiuk** [reply]
Well. To answer this question one would have to search through the repo by metric names. I suggest you do it yourself. The code is open to everyone.
